tools/cli/lib/api.c

```diff
@@ -1303,7 +1303,19 @@ uint32_t
 TDNFCliRefresh(
     PTDNF_CLI_CONTEXT pContext)
 {
-    return TDNFRefresh(pContext->hTdnf);
+    uint32_t dwError = 0;
+    dwError = TDNFRefresh(pContext->hTdnf);
+
+    if (dwError == ERROR_TDNF_SYSTEM_BASE + EACCES) {
+        if (geteuid()) {
```

Comment:
Use `gEuid` global variable here. If not please make adjustments so that `gEuid` is populated properly here. I think this can be avoided.

Reply:
Do you have a reason for this? `geteuid()` gives the same information (it's not changing), and the performance impact is negligible (if that is your concern).

Reply:
Why make a function call when we have a global variable which is dedicated for the exact thing. Redundancy is my concern. And performance as well, even though it is negligible.

Reply:
Okay, I thought about this - I'll leave it as is. Performance can be ignored here (we already spent more time discussing it than we will ever save). Having a global variable is actually adding redundance, and makes it more complex. For example, if there is an issue, we'd need to figure out if the variable is properly set and so on. But thank you for your comment.